presto-tests/src/main/java/io/prestosql/tests/DistributedQueryRunner.java

```diff
@@ -153,7 +153,7 @@ private DistributedQueryRunner(
         }
 
         // copy session using property manager in coordinator
-        defaultSession = defaultSession.toSessionRepresentation().toSession(coordinator.getMetadata().getSessionPropertyManager());
+        defaultSession = defaultSession.toSessionRepresentation().toSession(coordinator.getMetadata().getSessionPropertyManager(), defaultSession.getIdentity().getExtraCredentials());
```

Comment:
what makes extra credentials special? Are they not part of the session representation?

Reply:
They are not part of `SessionRepresentation` because that object is not secure and is available in places that shouldn't have the credentials. For example, it's part of `BasicQueryInfo` which is sent to the browser for the UI.